src/mapped_file.cpp

```diff
@@ -323,7 +323,7 @@ void mapped_file_impl::try_map_file(param_type p)
             access,
             (DWORD) (p.offset >> 32),
             (DWORD) (p.offset & 0xffffffff),
-            (SIZE_T) (size_ != max_length ? size_ : 0), 
+            (SIZE_T) (static_cast<size_type>(size_) != max_length ? size_ : 0),
```

Comment:
I'd suggest using `boost::numeric_cast<>` so that in the event there is actually an overflow it will be handled correctly with an exception.  Otherwise looks good.